### ISSUE TITLE
GraphCommerce demo site (https://graphcommerce.vercel.app/) is not accessible

### ISSUE THREAD

**Issue by dannya** [graphcommerce-org/graphcommerce#1995]
### Describe the Bug

GraphCommerce demo site (https://graphcommerce.vercel.app/) is not accessible.
Despite a first pass with the Lighthouse audit tool in Google Chrome resulting in a score of 100 for accessibility, this is misleading as proven by further manual testing.

Manual testing using keyboard-only navigation and the NVDA screen reader reveals some significant issues, especially with the header. There is no "skip to content" link before the header menu, and the tab ordering of the header top-level items is not correct for keyboard-only use. The "Shop" top-level section is completely inaccessible via keyboard - tabbing just skips past this item.

The "Speech bubble" and "Head" icons on the right both have a name of "Account" when using a screen reader - these should have unique names.

There are also visually-hidden items that are tabbed onto between the "Blog" item and the search box, and between the "Sign in" and "Cart" buttons.

### Expected Behavior

Site should be fully accessible using keyboard-only navigation and/or screen reader use.

This is all basic stuff that should work and have been caught already in your development process.

What is the level of accessibility awareness of your development team?

### To Reproduce

Try to use the mega menu using only a keyboard.
Try to use the mega menu non-visually, using a screen reader.

**Comment by paales** [comment]
Hi @dannya, thanks for the report! 

> This is all basic stuff that should work and have been caught already in your development process.

This is the first report about it, so it might not be so basic as it seems to you. You are probably more experienced in the accessibility stuff that some others.

We do like to get these things fixed, are you in the position to help with that? I can allocate some resources to get this fixed, can you help review the PR and help find a11y issues?

Todo:

- [ ] Add 'skip to content' button that is accessible by keyboard.
- [ ] Tab ordering of the header top-level items is not correct for keyboard-only use. The "Shop" top-level section is completely inaccessible via keyboard - tabbing just skips past this item.
- [ ] The "Speech bubble" and "Head" icons on the right both have a name of "Account" when using a screen reader - these should have unique names.
- [ ] There are also visually-hidden items that are tabbed onto between the "Blog" item and the search box, and between the "Sign in" and "Cart" buttons.

Any more basics we missed?

**Comment by dannya** [comment]
Hi @paales, I would be happy to help review the PR when it is ready - do you have a "staging" URL for the site to assist with this?

**Comment by dannya** [comment]
Some further issues:

- [ ] Search icon (magnifying glass) does not have a screen reader label, like `aria-label="Search..."` (like the hidden button next to it does - this is more of a technical fail, as it did read out "Search link" using NVDA screen reader, but you should add it to improve your Lighthouse audit score:

![image](https://github.com/graphcommerce-org/graphcommerce/assets/394687/d7d13609-3f6d-40fc-b14f-a6f56e4d88a3)

- [ ] Lots of interactable elements don't have a hover / focus visual state (aside from the cursor change) - classic case is the "Add to Wishlist" heart icon in the grid view items. It is usual to have a colour change or similar when hovering over items / links / etc.

- [ ] None of the cards in the grid view or carousel have a focus state, making navigation using a keyboard almost impossible.

These are just some more items from a cursory inspection, I'm also available on a consultancy basis if you'd like a more complete audit.

**Comment by FrankHarland** [comment]
Hi @dannya 👋 

Thank you for providing feedback on this important issue.
I'm by no means an WCAG / A11y expert, but made an effort to improve on the points mentioned in your issue.

- Added skip content link
- Removed empty buttons from tab flow
- Gave focus to elements (such as the menu) that appear when after clicking a button
- Improved aria labels where needed

From my end I'm now able to use a screen reader to "blindly" navigate the demo site and place my order.
If you would like to try so yourself please visit: https://graphcommerce-git-feature-gcom-1221-graphcommerce.vercel.app/en-gb 

Any further feedback is welcome.

You'll notice I have not made any visual modifications regarding hover/focus states as I'll need to run these by the design department first. While I think it's a good idea for us to make the demo site better in this regard, I'd like to point out that the framework has loads of options to style things as needed for your specific project.

**Comment by dannya** [comment]
Hi @FrankHarland , thanks for your update and detailed changes.

I've noticed there is still a tab accessible invisible element in your latest deployment link:

![image](https://github.com/graphcommerce-org/graphcommerce/assets/394687/9539d255-a595-407c-b120-2929cb283ffb)

**Comment by FrankHarland** [comment]
Hi @dannya 

It's been a while but I've just pushed some new changes to the [test environment](https://graphcommerce-git-feature-gcom-1221-graphcommerce.vercel.app/en-gb).
The invisible links should no longer be accesable via keyboard navigation.
I've also made improvements to other elements, such as the searchbar which can now be focused using keyboard navigation.
We also had a issues where the user could open a menu by pressing a button, but the focus would not be set on the newly openend menu or modal, this has also been resolved in various places.

**Comment by dannya** [comment]
@FrankHarland looks good, thanks for your effort here

**Comment by FrankHarland** [comment]
Thanks @dannya for reporting and testing this issue. I'll see to it that this gets merged and included in the next release.
